packages/propel/.storybook/main.ts

```diff
@@ -0,0 +1,20 @@
+import type { StorybookConfig } from "@storybook/react-vite";
+
+import { join, dirname } from "path";
+
+/**
+ * This function is used to resolve the absolute path of a package.
+ * It is needed in projects that use Yarn PnP or are set up within a monorepo.
+ */
+function getAbsolutePath(value: string): any {
```

Comment:
```suggestion
function getAbsolutePath(value: string) {
```

is `any` necessary? I feel like in recent versions of storybook, they've done a massive cleanup of types.

Reply:
I think instead of using require.resolve, we might be able to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta/resolve if this project is ESM.

Reply:
I tried import.meta.resolve first, but it caused a Storybook error because the .storybook/main.ts config runs in CommonJS mode, not ESM. 

The error was: `"import.meta" is not available with the "cjs" output format`

So require.resolve is the correct choice here for Storybook compatibility, even though import.meta.resolve would be preferred in a pure ESM environment.